app/src/main/java/com/github/sdpteam15/polyevents/view/activity/TimeTableActivity.kt

```diff
@@ -0,0 +1,510 @@
+     */
+    private fun addNowBar() {
+        val now = LocalDateTime.now()
+        val nowH = now.hour
+        val nowM = now.minute
+        val idLine = hourToLine[nowH] ?: return
+
+        val currentId = nextId++
+        val constraintLayout = findViewById<ConstraintLayout>(R.id.id_timetable_constraintlayout)
+
+        //Create now line
+        val lign = View(this)
+        lign.backgroundTintList = resources.getColorStateList(R.color.red, null)
+        lign.id = currentId
+
+        //Create new parameters for the line (the line)
+        val params = ViewGroup.LayoutParams(0, nowLineHeightDP.dpToPixelsInt(this))
+        lign.layoutParams = params
+        lign.setBackgroundColor(Color.BLUE)
+
+        constraintLayout.addView(lign)
+        idNowBar = lign
+
+
+        val constraintSet = ConstraintSet()
+        constraintSet.clone(constraintLayout)
+        val marginTop = ((nowM / 60.0) * hourSizeDp).toInt()
+        //Constraints for the line
+        constraintSet.connect(
+            currentId,
+            ConstraintSet.TOP,
+            idLine,
+            ConstraintSet.BOTTOM,
+            marginTop.dpToPixelsInt(this)
+        )
+        constraintSet.connect(
+            currentId,
+            ConstraintSet.LEFT,
+            idLine,
+            ConstraintSet.LEFT,
+            linepaddingLeftDP.dpToPixelsInt(this)
+        )
+        constraintSet.connect(
+            currentId,
+            ConstraintSet.RIGHT,
+            constraintLayout.id,
+            ConstraintSet.RIGHT,
+            0
+        )
+
+        //Apply all the constraints
+        constraintSet.applyTo(constraintLayout)
+
+
+        //TODO : Maybe call the runnable itself instead of recreating each time
+        val handler = Handler()
+        val task = Runnable { refreshNowBar() }
+        val period = (60 - now.second.toLong()) * 1000
+
+        handler.postDelayed(task, period)
+    }
+
+    /**
+     * Refreshes the display of the bar that indicates the current time
+     */
+    private fun refreshNowBar() {
+        idNowBar ?: return
+        val constraintLayout = findViewById<ConstraintLayout>(R.id.id_timetable_constraintlayout)
+        constraintLayout.removeView(idNowBar)
+        idNowBar = null
+        addNowBar()
+    }
+
+    override fun onItemSelected(p0: AdapterView<*>?, p1: View?, newPos: Int, p3: Long) {
+        if (selectedItem != newPos) {
+            selectedItem = newPos
+            selectedZone = obsZoneNames.keys.toList()[newPos]
+            drawZoneEvents(selectedZone!!)
+        }
+    }
+
+    override fun onNothingSelected(p0: AdapterView<*>?) {
+
+    }
+
+    /**
+     * Helper function to have size in pixels from dp
+     * https://android--code.blogspot.com/2020/08/android-kotlin-convert-dp-to-pixels.html
+     */
+    fun Int.dpToPixelsInt(context: Context): Int = TypedValue.applyDimension(
+        TypedValue.COMPLEX_UNIT_DIP, this.toFloat(), context.resources.displayMetrics
+    ).toInt()
```

Comment:
Interesting to do it like that.
I could have use this type of function creation before.